### PR TITLE
[NU-1616] Docker Compose based installation example

### PR DESCRIPTION
**Pull request by mk-software-pl**
## Describe your changes

* example installation of Nu ecosystem (it's not a production setup)

Running:
```bash
docker compose up -d --build
```
or
```bash
./run.sh
```

Cleaning:
```bash
docker compose down -v
```
or
```bash
./clean.sh
```

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: examples/installation/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      interval: 10s
+      retries: 10
+    volumes:
+      - ./grafana:/etc/grafana/provisioning
+      - ./grafana/dashboards:/var/lib/grafana/dashboards
+
+  influxdb:
+    image: influxdb:1.8.10
+    restart: unless-stopped
+    hostname: influxdb
```
Comment (mk-software-pl) [inline examples/installation/docker-compose.yml:98]: missing DB NAME

File: examples/installation/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,208 @@
+name: nussknacker
+
+services:
+
+  nginx:
```
Comment (mk-software-pl) [inline examples/installation/docker-compose.yml:5]: problem with access:
* localhost:8080/flink
* localhost:8080/akhq

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -242,3 +242,28 @@ Below you can find out some hints how to achieve that:
 - Prefer to use methods instead of member functions (def foo: (T) => R)
 - Avoid using AnyVal if the class is in API that will be used from Java
 - Prefer adding the `final` keyword to `case classes` definitions if you have no serious reason to skip it
+
+
+# Examples
```
Comment (arkadius) [inline CONTRIBUTING.md:247]: Contributing doc as name point, is for contributors, not for users. IMO it is not a good place to mention about that.

Comment (coutoPL) [reply CONTRIBUTING.md:247]: Yes, I know. We are going to mention that in the docs obviously. 
Do you think there is any point in informing contributors of this as well?

Comment (mk-software-pl) [reply CONTRIBUTING.md:247]: Yes, I know. We are going to mention that in the docs obviously.
Do you think there is any point in informing contributors of this as well?

Comment (arkadius) [reply CONTRIBUTING.md:247]: I don't think that informing contributors of this is necessary. They'll find examples in the docs. Does docs will be a part of this PR? e2e perspective would help to check if everything will fit together.

**Review by lukasz-bigorajski — APPROVED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (mk-software-pl) [reply CONTRIBUTING.md]: No, it won't be :D Documentation changes are a huge part of the epic and there are separate tasks for that.

File: examples/installation/flink/flink-properties.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# We can have many jobs per cluster, in such setting managed memory is not easy to tune
+state.backend.rocksdb.memory.managed: false
+# For frequent writes increase the value as needed. Currently RocksDB settings can only be changed per Flink cluster
+state.backend.rocksdb.writebuffer.size: 256m
+
+metrics.reporters: influxdb_reporter
+metrics.reporter.influxdb_reporter.factory.class: org.apache.flink.metrics.influxdb.InfluxdbReporterFactory
+metrics.reporter.influxdb_reporter.host: telegraf
+metrics.reporter.influxdb_reporter.port: 8087
+metrics.reporter.influxdb_reporter.db: esp
```
Comment (arkadius) [inline examples/installation/flink/flink-properties.yml:21]: What about that? We leave esp for now?

File: examples/installation/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      interval: 10s
+      retries: 10
+    volumes:
+      - ./nginx/nginx.conf:/etc/nginx/nginx.conf
+      - ./nginx/additional:/etc/nginx/additional
+
+  ### Nussknacker designer service
+
+  designer:
+    image: touk/nussknacker:1.14.0_scala-2.12
```
Comment (lciolecki) [inline examples/installation/docker-compose.yml:26]: Why scala 2.12? Let's do it with newer scala, wdyt?

File: examples/installation/nginx/nginx.conf [hunk 6]
```diff
@@ -0,0 +1,26 @@
+http {
+  server {
+    listen 8080;
+
+    location /grafana/ {
```
Comment (lciolecki) [inline examples/installation/nginx/nginx.conf:5]: To be consistent, let's move it to additional?

File: examples/installation/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+      GF_SECURITY_ALLOW_EMBEDDING: "true"
+    depends_on:
+      influxdb:
+        condition: service_healthy
+    healthcheck:
+      test: ["CMD-SHELL", "nc -z localhost 3000"]
+      interval: 10s
+      retries: 10
+    volumes:
+      - ./grafana:/etc/grafana/provisioning
```
Comment (lciolecki) [inline examples/installation/docker-compose.yml:92]: ./grafana/datasources:/etc/grafana/provisioning

File: examples/installation/flink/flink-properties.yml (hunk 4 above)
Comment (mk-software-pl) [reply examples/installation/flink/flink-properties.yml:21]: I'd like to cherry-pick this commit (after merging) to the `release/1.15` branch. So, the change with the changing default will be done in the next PR.

File: examples/installation/docker-compose.yml (hunk 5 above)
Comment (mk-software-pl) [reply examples/installation/docker-compose.yml:26]: I did it intentionally. Until you want to create custom components, you (as a user) don't care about Scala. At the moment Nu with Scala 2.13 is more complicated to install (due to the patching steps of Flink). That's why it's better to use 2.12 in the example. 

AFAIK we will address the problems with Scala 2.13 in the future.

File: examples/installation/nginx/nginx.conf (hunk 6 above)
Comment (mk-software-pl) [reply examples/installation/nginx/nginx.conf:5]: I'm not sure. Grafana is the required dependency, that why its routing is defined here.

File: examples/installation/docker-compose.yml (hunk 7 above)
Comment (mk-software-pl) [reply examples/installation/docker-compose.yml:92]: What is the advantage of this approaach?

File: examples/installation/flink/flink-properties.yml (hunk 4 above)
Comment (mk-software-pl) [reply examples/installation/flink/flink-properties.yml:21]: this PR: https://github.com/TouK/nussknacker/pull/6082

**Pull request merged**